app/service/api/v2/converter.rb

```diff
@@ -0,0 +1,28 @@
+class Api::V2::Converter
+  def to_api_v2_form_document(form_snapshot)
+    form_snapshot = form_snapshot.deep_dup
+
+    form_snapshot.delete("id")
+
+    pages = form_snapshot.delete("pages")
+    form_snapshot["steps"] = pages.map { |page| to_api_v2_step(page) }
+
+    form_snapshot
+  end
+
+private
+
+  def to_api_v2_step(page)
+    step = {
+      "id" => page["id"],
+      "position" => page["position"],
+      "next_step_id" => page["next_page"],
+    }
+    if page["question_text"].present?
+      step["type"] = "question_page"
+      step["data"] = page.slice(*%w[question_text hint_text answer_type is_optional answer_settings page_heading guidance_markdown is_repeatable])
```

Comment:
Interesting thought 🤔 It has been a bit of a pain having the nested `answer_settings`, and it wouldn't be too difficult to unwrap it here... I think it might make the forms-runner work more difficult though.

Thinking out loud, I guess the question to ask is if this is our best opportunity to make this change, or if there will be other opportunities later on 😅 It's definitely a breaking change, but it would maybe be possible to make it work in forms-runner if it supported both possibilities; then again we could avoid the extra complexity if we fold it in now.

I'll see what other devs think.

Reply:
@thomasiles do we have any tickets/notes somewhere describing the issues with the `answer_settings`?

Reply:
I've started a discussion on Slack: https://gds.slack.com/archives/C02QNDA5YBC/p1726037689099619

Reply:
re tickets/notes etc. I don't have anything to point at.

It's also not a blocker for anything in the runner, more of a style thing. There have been times when you have to take the extra step of checking asnwer_settings exists before accessing something. I think it may have been discussed when we were changing the API to rails.

I think it you are right to think about the best time to change it. If it makes the runner using API v2 harder and we can do it later, that might be our best bet.